smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/SwiftSettings.kt

```diff
@@ -38,7 +39,8 @@ class SwiftSettings(
     val sdkId: String,
     val gitRepo: String,
     val swiftVersion: String,
-    var useInterceptors: Boolean
+    var useInterceptors: Boolean,
+    val mergeModels: Boolean,
```

Comment:
The `SwiftSettings` type (which is itself a property of generation context) is where you can find the merge models setting.